File: api/download.py
import requests

def handler(request, response):
    url = request.query.get("url")
    if not url:
        response.status_code = 400
        return response.send("Missing 'url' query parameter")

    try:
        img_res = requests.get(url)
        if img_res.status_code != 200:
            response.status_code = 502
            return response.send("Failed to fetch image")

        response.set_header("Content-Type", img_res.headers.get("Content-Type", "application/octet-stream"))
        response.set_header("Content-Disposition", 'attachment; filename="enhanced.jpg"')
        response.body = img_res.content
        return response
    except Exception as e:
        response.status_code = 500
        return response.send(f"Error: {str(e)}")

File: api/download.js
export default async function handler(req, res) {
  const imageUrl = req.query.url;

  if (!imageUrl) {
    return res.status(400).send("Missing image URL");
  }

  try {
    const response = await fetch(imageUrl);
    if (!response.ok) {
      throw new Error("Failed to fetch image");
    }

    const contentType = response.headers.get("content-type") || "application/octet-stream";
    const buffer = await response.arrayBuffer();

    res.setHeader("Content-Type", contentType);
    res.setHeader("Content-Disposition", 'attachment; filename="enhanced.jpg"');
    res.send(Buffer.from(buffer));
  } catch (error) {
    res.status(500).send("Download failed: " + error.message);
  }
}
